app/models/piece.rb

```diff
@@ -156,6 +156,7 @@ def can_be_blocked?(new_x, new_y) # used for determining checkmate. Our piece is
         return true if piece.valid_move?(coords.first, coords.last)
       end
     end
```

Comment:
Same issue here with type: KING. Don't know why you need to query for pieces that are not kings, though I understand that kings can only move one square and therefore cannot be blocked from moving into a square.

Reply:
I started an infinite loop (an undesirable one!) when I tried to add the line to the king.rb file that checked for moving into check. I got around this by changing the query to skip kings.. but I'm going to undo my changes now.